ruby-package/lib/base_thrift_client.rb

```diff
@@ -12,6 +12,7 @@ def initialize(server, port)
         @server = server
         @port = port
         @executor = init_executor(init_protocol())
+        @max_retries = 3
```

Comment:
Good idea.  I felt like I shouldn't hard-code a constant like this, but a full config file felt too heavyweight. I think optional parameter is best.